compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/symbols/BSymbol.java

```diff
@@ -66,7 +68,12 @@ public BSymbol(int tag, int flags, Name name, PackageID pkgID, BType type, BSymb
         this.type = type;
         this.owner = owner;
         this.pos = pos;
-        this.origin = origin;
+
+        if (name.value.startsWith("$missingNode$")) {
```

Comment:
Can we do this check at the symbol enter phase? Then we can use the `BLangMissingNodesHelper`

Reply:
Can do. The concern is whether symbol enter is the only place where a `SOURCE` symbol can be created.

Reply:
I think it would be better to do it within this constructor if we can't ensure Pubudu's concern and this is always the common place

Reply:
IMO creator of the symbol should give the proper origin. 

Also, having to hardcode "$missingNode$" smells a problem

Reply:
Will move it to symbol enter and see. If any issues arise, can address them case by case I guess